prov/gni/include/gnix_cq.h

```diff
@@ -45,6 +45,9 @@
 #define GNIX_CQ_DEFAULT_FORMAT struct fi_cq_entry
 #define GNIX_CQ_DEFAULT_SIZE   256
 
+/* forward declaration */
+struct gnix_fid_ep;
```

Comment:
Include guards only work to a point. If 'a' includes 'b' and 'b' includes 'a', and both of them are header files, then it is possible that 'a' hasn't resolved by the time that 'b' includes 'a', meaning that the first include guard hasn't closed. There is a solid need for forward declarations for that reason, or at minimum, abstract typing.